src/views/compute/wizard/TemplateIsoSelection.vue

```diff
@@ -232,6 +235,57 @@ export default {
     },
     changeFilterType (value) {
       this.filterType = value
+    },
+    getTemplatesByKeyword (keyword) {
+      let apiCommand = ''
+      let apiCommandResponse = ''
+      this.filteredItems = []
+      const promises = []
+      const templates = []
+      const params = {}
+      const templateFilter = [
+        'featured',
+        'community',
+        'selfexecutable',
+        'sharedexecutable'
+      ]
+      // Fetch templates or isos depending on active tab
+      params.zoneid = _.get(this.zone, 'id')
+      params.keyword = keyword
+      if (this.inputDecorator === 'templateid') {
+        apiCommand = 'listTemplates'
+        apiCommandResponse = 'listtemplatesresponse.template'
+      } else {
+        apiCommand = 'listIsos'
+        apiCommandResponse = 'listisosresponse.iso'
+        params.bootable = 'true'
+      }
+      // Fetch templates/isos for all filters
+      templateFilter.forEach(filter => {
+        params.templatefilter = filter
+        promises.push(this.fetchTemplatesIsos(apiCommand, keyword, params))
+      })
+
+      Promise.all(promises).then(response => {
+        response.forEach((resItem) => {
+          const concatTemplates = _.concat(templates, _.get(resItem, apiCommandResponse, []))
```

Comment:
@ravening with latest 4.14 that is what we use reference env, templates returned are unque when the API is called with a `showunique=true` param. This will not be necessary then, also use of lodash.

Reply:
if `_.concat(templates, _.get(resItem, apiCommandResponse, []))` is not needed then how can I extract response from `resItem`?